usl_pipeline/cloud_functions/main.py

```diff
@@ -307,12 +308,38 @@ def _read_wps_features(fd: IO[bytes]) -> Tuple[NDArray, FeatureMetadata]:
     # Ignore type checker error - BytesIO inherits from expected type BufferedIOBase
     # https://shorturl.at/lk4om
     with xarray.open_dataset(fd) as ds:  # type: ignore
-        feature = ds.data_vars["SNOALB"].values
-        # Drop time axis - each WPS output file will correspond to single time value
-        feature = numpy.squeeze(feature, axis=0)
+        features_matrix = numpy.array([])
+        for var in wps_variables.REQUIRED_VARS:
+            try:
+                feature = ds.data_vars[var].values
+                feature = _process_wps_feature(feature)
+
+                if not features_matrix.size:
+                    features_matrix = feature
+                else:
+                    features_matrix = numpy.dstack((features_matrix, feature))
```

Comment:
might refactor this as
```python
feature_components = []
for var in wps_variables.REQUIRED_VARS:
  feature = ds.data_vars[var].values
  ...
  feature_components.append(feature)

feature_matrix = numpy.dstack(feature_components)
```
I'd wager that a single dstack will be faster than repeated invocations.